### PR TITLE
Migrate additional website content

### PR DESCRIPTION
**Pull request by speth**
**Changes proposed in this pull request**

This PR transfers most of the remaining version-specific content from the [cantera-website](https://github.com/Cantera/cantera-website) into this repository. Specifically, this includes:

- Release notes
- Installation instructions
- C++ compilation and application tutorials
- Glossary

In addition, a new page has been added to the "Develop" section explaining a bit about how the source material for the documentation is organized and some useful syntax for working with MyST, reST, and Doxygen.

**If applicable, fill in the issue number this pull request is fixing**

Continues work on https://github.com/Cantera/enhancements/issues/178

**If applicable, provide an example illustrating new features this pull request is introducing**

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: doc/doxygen/mainpage.md
Comment (bryanwweber) [inline doc/doxygen/mainpage.md]:
```suggestion
Use the sidebar to view detailed documentation of the C++ code, or use the following
```

File: doc/doxygen/mainpage.md
Comment (bryanwweber) [inline doc/doxygen/mainpage.md]:
```suggestion
Details of the fundamental scientific theory used in the implementation of %Cantera are located in the
```

File: doc/sphinx/develop/doc-formatting.md
Comment (bryanwweber) [inline doc/sphinx/develop/doc-formatting.md]:
```suggestion
is generated by [Sphinx](https://sphinx-doc.org). The C++ API documentation is generated using Doxygen. The
```

File: doc/sphinx/develop/doc-formatting.md
Comment (bryanwweber) [inline doc/sphinx/develop/doc-formatting.md]:
```suggestion
- Caution on usage of single backticks versus double backticks -- the former are only for linking to things using the default role (that is, Python objects)
```

File: doc/sphinx/install/pip.md
Comment (bryanwweber) [inline doc/sphinx/install/pip.md]:
```suggestion
  and restoring `SolutionArray` and flame objects to/from HDF5 files are not available.
```

File: doc/sphinx/install/ubuntu.md
Comment (bryanwweber) [inline doc/sphinx/install/ubuntu.md]: I wonder if it's worth adding something like, non-LTS versions are supported on a best-effort basis and in general for X timeframe?

File: doc/sphinx/reference/glossary.md
Comment (bryanwweber) [inline doc/sphinx/reference/glossary.md]:
```suggestion
: Product of the universal gas constant (R) and the temperature
```

File: doc/sphinx/reference/thermo/index.md
Comment (bryanwweber) [inline doc/sphinx/reference/thermo/index.md]:
```suggestion
both levels, and these selections must be compatible with one another. For instance, one
```

File: doc/sphinx/userguide/compiling-cxx.md
Comment (bryanwweber) [inline doc/sphinx/userguide/compiling-cxx.md]: Link to CMake?

File: doc/sphinx/userguide/compiling-cxx.md
Comment (bryanwweber) [inline doc/sphinx/userguide/compiling-cxx.md]: Link to SCons?

**Review by bryanwweber — DISMISSED** [review]
A few suggestions, but lgtm! Merge whenever you want

**Comment by speth** [comment]
Thanks for looking this over, @bryanwweber. I made the suggested changes, which requires re-approval before I can merge.

File: doc/sphinx/userguide/demo1a.cpp
Comment (ischoegl) [inline doc/sphinx/userguide/demo1a.cpp:1]: I believe the demo C++ files should be moved to `samples/cxx`, so they can be part of CI?

Same for all other `cpp` files in this folder.

File: doc/sphinx/userguide/python-tutorial.md
Comment (ischoegl) [inline doc/sphinx/userguide/python-tutorial.md:203]: Is this a typo? (empty `[]`). Similar below.

**Review by ischoegl — COMMENTED** [review]
@speth ... thank you for your continued efforts on this. 

I mainly have a comment on the location of the demo C++ files. I believe that they should be moved to `samples/cxx` so they are covered by CI, which will ensure that they are kept up to date. Whether this is done in this PR or a subsequent one is inconsequential, as long as it is taken care of before the release of 3.1.

**Review by bryanwweber — DISMISSED** [review]
Thanks @speth!

File: doc/sphinx/userguide/python-tutorial.md
Comment (speth) [reply doc/sphinx/userguide/python-tutorial.md:203]: No, this is the syntax for automatically generating the link text based on the title of the linked page.

**Review by ischoegl — DISMISSED** [review]
Thanks, @speth, feel free to merge. My remaining comment can be easily taken care of in a follow-up PR.

**Comment by speth** [comment]
> I mainly have a comment on the location of the demo C++ files. I believe that they should be moved to `samples/cxx` so they are covered by CI, which will ensure that they are kept up to date. Whether this is done in this PR or a subsequent one is inconsequential, as long as it is taken care of before the release of 3.1.

I was planning on setting these up to run as part of the test suite but forgot before opening this PR. So, rather than wait and risk forgetting to ever do this, it is now implemented. I went with an option that's slightly different from your suggestion, since I only want to include these samples in the C++ Tutorial and not have them repeated in the Examples section, where I think they would distract from the more interesting example programs.

**Review by bryanwweber — APPROVED** [review]

**Pull request merged**
